src/mds/mdstypes.cc

```diff
@@ -431,6 +431,12 @@ void feature_bitset_t::decode(bufferlist::const_iterator &p) {
   }
 }
 
+void feature_bitset_t::dump(Formatter *f) const {
+  CachedStackStringStream css;
```

Comment:
Need to include `common/StackStringStream.h`.

Reply:
Was this required for a build failure? Works for me (w/o the include) since the header gets included indirectly.

Guess, we should be fixing other indirect header includes too.

Reply:
Normally it's included via `dout.h` but I guess this file doesn't do that.